bonfire/bonfire.py

```diff
@@ -903,65 +914,115 @@ def _cmd_process(
     print(json.dumps(processed_templates, indent=2))
 
 
-def _get_namespace(requested_ns_name, name, requester, duration, pool, timeout, local, force):
+def _get_namespace(
+    requested_ns_name,
+    name,
+    requester,
+    duration,
+    pool,
+    timeout,
+    local,
+    force,
+    using_current=False
+):
     reserved_new_ns = False
 
     if not has_ns_operator():
         if requested_ns_name:
             ns = Namespace(name=requested_ns_name)
+            return ns.name, requested_ns_name
         else:
             _error(f"{NO_RESERVATION_SYS}. Use '-n' to provide a specific target namespace")
 
-    else:
-        if requested_ns_name:
-            log.debug(
-                "checking if namespace '%s' has been reserved via ns operator...", requested_ns_name
-            )
-            operator_reservation = get_reservation(namespace=requested_ns_name)
-            if not operator_reservation:
-                _error(f"Namespace '{requested_ns_name}' has not been reserved yet")
+    ns = None
+    if requested_ns_name:
+        ns = _check_and_use_namespace(requested_ns_name, using_current)
+
+    if not ns:
+        if using_current:
+            log.info("Current namespace '%s' could not be used (not reserved,"
+                     " expired or not owned), reserving a new one", requested_ns_name)
+
+        ns = _check_and_reserve_namespace(name, requester, duration, pool, timeout, local, force)
+        reserved_new_ns = True
+
+    return ns.name, reserved_new_ns
+
+
+def _check_and_use_namespace(requested_ns_name, using_current):
+    if not has_ns_operator():
+        _error(f"{NO_RESERVATION_SYS}")
+
+    log.debug(
+        "checking if namespace '%s' has been reserved via ns operator...", requested_ns_name
+    )
+    operator_reservation = get_reservation(namespace=requested_ns_name)
+    ns = None
+    if operator_reservation:
+        log.debug("found existing ns operator reservation")
+
+        if operator_reservation.get("status", {}).get("state") == "expired":
+            msg = f"Reservation has expired for namespace '{requested_ns_name}'"
+            if using_current:
+                log.info(msg)
+                return None
             else:
-                log.debug("found existing ns operator reservation")
+                _error(msg)
 
-                if operator_reservation.get("status", {}).get("state") == "expired":
-                    _error(f"Reservation has expired for namespace '{requested_ns_name}'")
+        ns = Namespace(name=requested_ns_name)
+        if not ns.owned_by_me:
+            if using_current:
+                log.info("Current namespace '%s' is reserved by someone else, skipping",
```

Comment:
"skipping" confused me a bit. I see how it works now -- looks like we'll get the log msg from line 943 printed next. I think let's just remove the "skipping" part from this log message.

```suggestion
                log.info("current namespace '%s' is reserved by someone else",
```